flink-core/src/main/java/org/apache/flink/configuration/ConfigUtils.java

```diff
@@ -96,9 +96,7 @@
 				.filter(Objects::nonNull)
 				.collect(Collectors.toCollection(ArrayList::new));
 
-		if (!encodedOption.isEmpty()) {
-			configuration.set(key, encodedOption);
-		}
+		configuration.set(key, encodedOption);
```

Comment:
Since we remove the empty check here, then the `pipeline.classpaths` with empty value could be written in Flink configuration. And it will cause an `error` in the JobManager/TaskManager log.

```
2020-04-29 13:33:02,570 WARN  org.apache.flink.configuration.GlobalConfiguration           [] - Error while trying to split key and value in configuration file ./flink-conf.yaml:17: "pipeline.classpaths: "
```

Reply:
This is a warning, right? It will not cause any problem apart from polluting the logs. Is this correct?

Reply:
Yes, you are right. It will not cause any problem except for the log errors.